scss/4-elements/_typography.scss

```diff
@@ -1,59 +1,62 @@
 %cads-h1 {
   font-size: 2.5rem;
-  line-height: 1.2em; // Line-height is variable
+  line-height: 3.25rem;
   font-weight: $cads-font-weight__bold;
   color: $cads-font-colour__headings;
   margin: 0 0 $cads-spacing-7 0;
 
   @include cads-media-breakpoint-down(sm) {
     font-size: 2rem;
+    line-height: 2.5rem;
     margin: 0 0 $cads-spacing-5 0;
   }
 }
 
 %cads-h2 {
   font-size: 2rem;
-  line-height: 1.25em; // Line-height is variable
+  line-height: 2.75rem;
   font-weight: $cads-font-weight__bold;
   color: $cads-font-colour__headings;
   margin: 0 0 $cads-spacing-4 0;
 
   @include cads-media-breakpoint-down(sm) {
     font-size: 1.5rem;
+    line-height: 2rem;
     margin: 0 0 $cads-spacing-3 0;
   }
 }
 
 %cads-h3 {
-  font-size: 1.4rem;
-  line-height: 1.25em; // Line-height is variable
+  font-size: 1.5rem;
+  line-height: 2rem;
   font-weight: $cads-font-weight__bold;
   color: $cads-font-colour__headings;
-  margin: 0 0 $cads-spacing-3 0;
+  margin: 0 0 $cads-spacing-4 0;
 
   @include cads-media-breakpoint-down(sm) {
     font-size: 1.25rem;
-    margin: 0 0 $cads-spacing-3 0;
+    line-height: 1.75rem;
   }
 }
 
 %cads-h4 {
   font-size: 1.125rem;
-  line-height: 1.5rem; // Line-height remains the same
+  line-height: 1.625rem;
   font-weight: $cads-font-weight__bold;
   color: $cads-font-colour__headings;
-  margin: 0 0 $cads-spacing-3 0;
+  margin: 0 0 $cads-spacing-4 0;
 
   @include cads-media-breakpoint-down(sm) {
     font-size: 1rem;
+    line-height: 1.625rem;
   }
 }
 
 %cads-paragraph {
   @include cads-typographic-scale-text();
 
   font-weight: $cads-font-weight__regular;
-  margin: 0 0 $cads-spacing-4 0;
+  margin: 0 0 $cads-spacing-5 0;
 }
 
 %cads-paragraph-small {
```

Comment:
The small paragraph should have the same margin specs as the normal paragraph (24px margin-bottom) on all the breakpoints.